File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int l = 0, r = nums.size()-1;
        int area = 0;
        while(l<r){
            int w = r-l;
            area = max(area, w* min(nums[l],nums[r]));
            if(nums[l]>nums[r]){
                r--;
            }else if(nums[l]<nums[r]){
                l++;
            }
            else{
                l++;
                r--;
            }
        }
        return area;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        water = 0
        while l < r:
            width = r - l
            h = min(height[l], height[r])
            water = max(water, width*h)
            if height[l] > height[r]:
                r -= 1
            else:
                l += 1
        return water



        